packages/react/src/components/List/Lists.tsx

```diff
@@ -1,66 +1,56 @@
 import { Slot } from '@radix-ui/react-slot';
 import cl from 'clsx/lite';
-import type { HTMLAttributes, OlHTMLAttributes } from 'react';
-import { forwardRef, useContext } from 'react';
+import { forwardRef } from 'react';
 
 import { Paragraph } from '../Typography';
 
-import { ListContext } from './ListRoot';
-
-export type ListUnorderedProps = {
+type ListBaseProps = {
   /**
    * Change the default rendered element for the one passed as a child, merging their props and behavior.
    * @default false
    */
   asChild?: boolean;
-} & Omit<HTMLAttributes<HTMLUListElement>, 'size'>;
-
-export const Unordered = forwardRef<HTMLUListElement, ListUnorderedProps>(
-  ({ asChild, className, ...rest }, ref) => {
-    const { size, headingId } = useContext(ListContext);
-
-    const Component = asChild ? Slot : 'ul';
-
-    return (
-      <Paragraph size={size} asChild>
-        <Component
-          className={cl(`ds-list`, `ds-list--${size}`, className)}
-          {...(headingId ? { 'aria-labelledby': headingId } : {})}
-          ref={ref}
-          {...rest}
-        />
-      </Paragraph>
-    );
-  },
-);
-
-Unordered.displayName = 'ListUnordered';
-
-export type ListOrderedProps = {
   /**
-   * Change the default rendered element for the one passed as a child, merging their props and behavior.
-   * @default false
+   * Changes text sizing
+   * @default md
+   *
    */
-  asChild?: boolean;
-} & Omit<OlHTMLAttributes<HTMLOListElement>, 'size'>;
-
-export const Ordered = forwardRef<HTMLOListElement, ListOrderedProps>(
-  ({ asChild, className, ...rest }, ref) => {
-    const { size, headingId } = useContext(ListContext);
-
-    const Component = asChild ? Slot : 'ol';
-
-    return (
-      <Paragraph size={size} asChild>
-        <Component
-          className={cl(`ds-list`, `ds-list--${size}`, className)}
-          {...(headingId ? { 'aria-labelledby': headingId } : {})}
-          ref={ref}
-          {...rest}
-        />
-      </Paragraph>
-    );
+  size?: 'sm' | 'md' | 'lg';
+};
+
+export type ListUnorderedProps = ListBaseProps &
+  Omit<React.HTMLAttributes<HTMLUListElement>, 'size'>;
+
+export type ListOrderedProps = ListBaseProps &
+  Omit<React.OlHTMLAttributes<HTMLOListElement>, 'size'>;
+
+const render = <HTMLType extends HTMLElement>(
+  tagName: string,
+  { asChild, className, size = 'md', ...rest }: ListOrderedProps,
+  ref: React.ForwardedRef<HTMLType>,
```

Comment:
We have used `T` as name for generics in our codebase as is common in most programming languages that support this.

```suggestion
const render = <T extends HTMLElement>(
  tagName: string,
  { asChild, className, size = 'md', ...rest }: ListOrderedProps,
  ref: React.ForwardedRef<T>,
```

Reply:
🙈 Will do - I know this is the most common Typescript convention (as well as with other letters if needing several variables), but I do really not like it at all - we are preaching good variable names in all programming, but in Typescript `T` is the way to go? 😅 But, not that important - I'll update! 😊